application/common/models/User.php

```diff
@@ -292,18 +282,9 @@ public function fields(): array
             'locked',
         ];
 
-        if ($this->hasPasswordAlready()) {
-            $fields['password_expires_at_utc'] = function () {
-                return $this->getPasswordExpiration();
-            };
-
-            $fields['password_last_changed'] = function () {
-                /** @var $mostRecentPassword PasswordHistory */
-                $mostRecentPassword = $this->getPasswordHistories()
-                                           ->orderBy(['id' => SORT_DESC])
-                                           ->one();
-
-                return $mostRecentPassword->created_utc;
+        if ($this->password_id !== null) {
```

Comment:
Out of curiosity, was there a reason to go back from `hasPasswordAlready()` to `$this->password_id !== null`? The function name seemed more readable to me, both for code internal to this class and for external code, but you often have good reasons for your changes, so I thought I'd ask.